### PR TITLE
feat: [CDS-85044]: Create Open API for Re-run and Retry pipeline for v1

### PR DESCRIPTION
**Pull request by utkarshchoubey-harness**
Create Open API for Re-run and Retry pipeline for v1

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)

### CODE REVIEW

**Review by mankrit-singh — APPROVED** [review]

**Comment by utkarshchoubey-harness** [comment]
trigger uts

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -985,12 +985,108 @@ paths:
...
+        - $ref: '#/components/parameters/AccountHeader'
+        - schema:
+            type: string
+          in: query
+          name: module
+          description: 'Module type for the entity. If its from deployments,type will be CD , if its from build type will be CI'
+        - schema:
+            type: string
+          in: query
+          name: plan_execution_id
```
Comment (shalini-agr) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1056]: can we make this a path param like in the rerun API above?

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 988 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2nhtB4lN23QJms20:-Nk-2nhtB4lN23QJms21:b-xxq2hx) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L988)):*
> ```YAML
>       tags:
>         - Input Sets
>   '/v1/orgs/{org}/projects/{project}/pipelines/{pipeline}/rerun/{execution-id}':
> ```

do we use hyphen or underscore?



_Code quote:_
```YAML
execution-id}
```
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1036 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2pf-16VhR8in2I1z:-Nk-2pf-16VhR8in2I2-:bnj7hl0) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L1036)):*
> ```YAML
>         - Pipeline Execution
>       x-internal: true
>   '/v1/orgs/{org}/projects/{project}/pipelines/{pipeline}/retry/{execution-id}':
> ```

same as above
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3516 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2u_L8gCZvQJ8JUCn:-Nk-2u_L8gCZvQJ8JUCo:b-y3fl74) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L3516)):*
> ```YAML
>       type: object
>       properties:
>         inputset_pipeline_yaml:
> ```

@brijeshdhakar lets suggest better names for all params inside new apis

**Comment by brijeshdhakar** [comment]
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3516 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2u_L8gCZvQJ8JUCn:-Nk-pPi8CYdn_wPn9DgB:b10qdrn) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L3516)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@brijeshdhakar lets suggest better names for all params inside new apis
</blockquote></details>

Lets call it yaml same as pipeline-execute api.

**Comment by brijeshdhakar** [comment]
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 988 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2nhtB4lN23QJms20:-Nk0GNHTBXHWsmrOnf-Q:b-nshgxn) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L988)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

do we use hyphen or underscore?
</blockquote></details>

We are using hyphen only at all places in path params.

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3516 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2u_L8gCZvQJ8JUCn:-NkOwNldETIJ0ngFDpd2:b-pb1brm) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L3516)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

Lets call it yaml same as pipeline-execute api.
</blockquote></details>

inputs
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 4167 at r4](https://reviewable.io/reviews/harness/harness-core/55882#-NkOwU2693VZod5PAlBY:-NkOwU2693VZod5PAlBZ:b-gqpwi4) ([raw file](https://github.com/harness/harness-core/blob/0d4badeedcb2715312c9b87beb290df2f3dd9630/pipeline-service/contracts/openapi/v1/openapi.yaml#L4167)):*
> ```YAML
>           x-stoplight:
>             id: 51p0t57s170jm
>           description: InputSet YAML if the pipeline contains runtime inputs. This will be empty by default if pipeline does not contains runtime inputs
> ```

InputsSet -> Inputs 



_Code quote:_
```YAML
InputSet
```

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by utkarshchoubey-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)* status: 2 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3516 at r3](https://reviewable.io/reviews/harness/harness-core/55882#-Nk-2u_L8gCZvQJ8JUCn:-NkPTBji4SvWgV2eofbq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5bc32aac7346c541aff5a5f331219addbe45084e/pipeline-service/contracts/openapi/v1/openapi.yaml#L3516)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

inputs
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 4167 at r4](https://reviewable.io/reviews/harness/harness-core/55882#-NkOwU2693VZod5PAlBY:-NkPTC899BPRpJLezIxf:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0d4badeedcb2715312c9b87beb290df2f3dd9630/pipeline-service/contracts/openapi/v1/openapi.yaml#L4167)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

InputsSet -> Inputs
</blockquote></details>

Done.

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by utkarshchoubey-harness** [comment]
trigger messagecheck

**Review by adivishy1 — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55882)* status: 1. unresolved discussions.

**Comment by utkarshchoubey-harness** [comment]
trigger review

**Review by vtxorxwitty — APPROVED** [review]

**Review by mankrit-singh — APPROVED** [review]

**Pull request merged**